pkg/cmd/attestation/verify/verify.go

```diff
@@ -256,23 +256,29 @@ func runVerify(opts *Options) error {
 	}
 	attestations = filteredAttestations
 
-	policy, err := buildVerifyPolicy(opts, *artifact)
+	opts.Logger.VerbosePrintf("Verifying attestations with predicate type: %s\n", opts.PredicateType)
+
+	ec, err := newEnforcementCriteria(opts, *artifact)
```

Comment:
imho:
- this should happen before we even load the bundles. The PredicateType in FilterAttestations should be coming from the `ec`, not from `opts`
- like, even before we digest the artifact. nothing in the EC depends on the artifact or a network call, right?
- and _then_ right after we call newEnforcementCriteria we should call `err := Valid()` or w/e.

here I am thinking that `ec.Validate()` is separate from `newEC` only in so far that it might be easier to check for invalid states separately from setting them the flags.

for example, if the OIDCIssuer/Certificate.Issuer is empty on the `Valid()` call then that's an error! but it might not be obvious where to stick that check in the `new` func.

Reply:
I ended up creating a separate `EnforcementCriteria#Valid` method for checking that all required fields are set. It felt a little too crowded doing it in the constructor.